src/GeneratorState.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef POWER_WHISPERPOWER_GENERATORSTATE_HPP
+#define POWER_WHISPERPOWER_GENERATORSTATE_HPP
+
+namespace power_whisperpower {
+    /**
+     * @brief The state of the WhisperPower generator
+     *
+     */
+    enum GeneratorState {
+        RUNNING,
+        STOPPED,
+        STOPPING,
+        STARTING,
+        FAILURE
```

Comment:
What is the semantic of this one ?

The generator might be STOPPING and FAILURE at the same time. I think you need to really document what this means.

Reply:
I would actually recommend leaving it out of this enum, but adding another field to indicate failure.